### PR TITLE
Test that the GPU Poisson solver solution is numerically divergence-free

### LINKED ISSUE #200: GPU Poisson solver tests
Kind of scary that we don't have tests for this. I wrote some as part of PR #167 but commented them out as I couldn't get them to work (I kept getting some CUDA error). I should make sure to finish that off.

### PR DESCRIPTION
**Pull request by ali-ramadhan**
Until the CPU and GPU Poisson solvers are unified this should give us confidence that the GPU Poisson solver is doing its job.

Found it easier to write the test independently so this supersedes PR #238 

Resolves #200

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -213,6 +213,18 @@ float_types = (Float32, Float64)
             end
         end
 
+        @testset "Divergence-free solution [GPU]" begin
+            println("    Testing divergence-free solution [GPU]...")
+            @hascuda begin
+                for ft in float_types
```
Comment (glwagner) [inline test/runtests.jl]: btw, It's an almost-universally adopted julia convention to denote a 'type' with capital letters (`T`, `FT`, etc).

**Comment by ali-ramadhan** [comment]
> Much relieved to see this test. I'm sad that my old test that used dispatch had some bug, but it's academic at this point since we hope to unify the solvers soon.

I think you just had to permute the indices but yeah, and we can go back and use your test once they're unified.

File: test/runtests.jl (hunk 1 above)
Comment (ali-ramadhan) [reply test/runtests.jl]: True. I'll use `FT` in the future, and modify `runtests.jl` when I get the chance.

**Comment by glwagner** [comment]
Oh. I missed the fact that you have to permute both on input and output. Got it.

**Pull request merged**